File: 066_PlusOne/66.swift

class Solution {
    func plusOne(_ digits: [Int]) -> [Int] {
        
        var digits = digits
        var currentIndex = digits.count - 1
        
        while currentIndex>=0 && digits[currentIndex] == 9 {
            digits[currentIndex] = 0
            currentIndex -= 1
        }
        
        if currentIndex >= 0 {
            digits[currentIndex] += 1
        } else {
            digits.insert(1, at: 0)
        }
        
        return digits
    }
}

File: 0066_PlusOne/66.cpp
#include <iostream>
#include <vector>
#include "../overload/ostream_ext.h"

using namespace std;

class Solution
{
public:
    vector<int> plusOne(vector<int> &digits)
    {
        int index = digits.size() - 1;

        while (index >= 0 && digits[index] == 9)
        {
            digits[index--] = 0;
        }

        if (index >= 0)
        {
            digits[index]++;
        }
        else
        {
            digits.insert(digits.begin(), 1);
        }

        return digits;
    }
};

int main()
{
    vector<int> q = {9};
    Solution s = Solution();
    auto ans = s.plusOne(q);
    cout << ans << endl;
    return 0;
}